docs/docs/authentication-and-access-control/social-auth.md

```diff
@@ -324,6 +324,67 @@ export class AuthController {
 }
 ```
 
+## Enable Code Flow with PKCE
```

Comment:
This section is before `## Custom Provider` on the same level. Could we move it into `## Custom Provider` as `### Enable Code Flow with PKCE` just before the official providers as you suggested? 🙂

Reply:
Done.

I added a note in twitter provider documentation. I found that twitter has a oauth2 settings to set the app to be used with a public client or a confidential client, and if you choose confidential client don't work as expected because tokenEndpoint expect a bearer token and abstract provider doesn't add it. What you think about this? we leave it as a limitation or maybe we can think in a solution? I don't know how many people use this with a confidential client.

Reply:
The social provider architecture in FoalTS implies that clients are confidential. The request that exchanges the auth code against an access token is not made by the browser or mobile in our case but by the server itself. If we choose a public client, then the client secret is simply ignored and we lost in security. I added the feature in the abstract provider so than can use confidential clients 👍